platforms/nuttx/src/px4/microchip/mpfs/version/board_mcu_version.c

```diff
@@ -187,20 +266,57 @@ int board_get_px4_guid_formated(char *format_buffer, int size)
  ************************************************************************************/
 int board_determine_hw_info(void)
 {
-	determine_hw();
+	struct system_version_string_s ver_str;
+	struct system_version_s ver;
+	orb_advert_t ver_str_pub = orb_advertise(ORB_ID(system_version_string), NULL);
+	orb_advert_t ver_pub = orb_advertise(ORB_ID(system_version), NULL);
+
+	uint32_t fpga_version = getreg32(FPGA_VER_REGISTER); // todo: replace eventually with device_boot_info
 
-	snprintf(hw_info, sizeof(hw_info), HW_INFO_INIT_PREFIX HW_INFO_SUFFIX HW_INFO_FPGA_PREFIX HW_INFO_FPGA_SUFFIX,
+	memset(&ver_str, 0, sizeof(ver_str));
+	memset(&ver, 0, sizeof(ver));
+
+	determine_hw(fpga_version);
+
+	snprintf(hw_info, sizeof(hw_info), HW_INFO_INIT_PREFIX HW_INFO_SUFFIX " " HW_INFO_FPGA_PREFIX HW_INFO_FPGA_SUFFIX,
 		 hw_version, hw_revision, fpga_version_major, fpga_version_minor);
 
+	/* HW version */
+
+	snprintf(ver_str.hw_version, sizeof(ver_str.hw_version), HW_INFO_INIT_PREFIX HW_INFO_SUFFIX, hw_version, hw_revision);
+	ver.hw_version = fpga_version;
+
+	/* PX4 version */
+
+	strncpy(ver_str.sw_version, PX4_GIT_TAG_STR, sizeof(ver_str.sw_version));
+	ver.sw_version = parse_tag_to_version(PX4_GIT_TAG_STR);
+
+	/* NuttX version */
+
+	snprintf(ver_str.os_version, sizeof(ver_str.os_version), NUTTX_GIT_TAG_STR "-g" NUTTX_GIT_VERSION_STR);
+	ver.os_version = parse_tag_to_version(ver_str.os_version);
+
+	/* Bootloader version */
+
+	strncpy(ver_str.bl_version, device_boot_info.bl_version, sizeof(ver_str.os_version));
```

Comment:
Is here copy paste error ? sizeof `ver_str.os_version` should be `ver_str.bl_version` ?

Reply:
Yes! Very much so need to fix that.

I was also wondering if it makes more sense to better align with formats in src/lib/version/version.c , and not to invent an own encoding scheme for the binary format things. And maybe move the publishing directly in there as well. I will move this to Draft for now, and try out some different things first, and not jump directly in merging this one.

Reply:
Tried to move the publishing to common code; but versions related code is so messed up that it doesn't make sense to try to improve it really.

This is good for us, so let's just keep it the way it is.